Flaky account stream test

`test_accountStatus_twoStreams` sometimes fails. But only sometimes. There is something nondeterministic about account status streaming with the replaying mocks.

`Test Case '-[CanopyTests.ContainerAPITests test_accountStatus_twoStreams]' started.
/Users/runner/work/Canopy/Canopy/Targets/Canopy/Tests/ContainerAPITests.swift:136: error: -[CanopyTests.ContainerAPITests test_accountStatus_twoStreams] : XCTAssertEqual failed: ("[__C.CKAccountStatus, __C.CKAccountStatus, __C.CKAccountStatus]") is not equal to ("[__C.CKAccountStatus, __C.CKAccountStatus, __C.CKAccountStatus]")
Test Case '-[CanopyTests.ContainerAPITests test_accountStatus_twoStreams]' failed (1.031 seconds).`

--- a/Targets/Canopy/Sources/CKContainerAPI/CKAccountChangedSequence.swift
+++ b/Targets/Canopy/Sources/CKContainerAPI/CKAccountChangedSequence.swift
@@ -39,6 +39,10 @@ struct CKAccountChangedSequence: AsyncSequence {
         guard mockElementsToProduce > 0 else {
           return nil
         }
+        // Don’t emit the signal too quickly.
+        // Otherwise this was causing some flaky tests and out-of-order status delivery.
+        // A better solution would be to better sequence the task creation in ReplayingMockCKContainer.
+        try? await Task.sleep(nanoseconds: UInt64(0.01 * Double(NSEC_PER_SEC)))
         mockElementsToProduce -= 1
         return ()
       case let .live(notifications):

--- a/Targets/Canopy/Tests/ContainerAPITests.swift
+++ b/Targets/Canopy/Tests/ContainerAPITests.swift
@@ -87,6 +87,12 @@ final class ContainerAPITests: XCTestCase {
   }
   
   func test_accountStatus_stream() async {
+    // This test was sometimes failing. The cause was that the account statuses were sometimes delivered out of order.
+    // https://github.com/tact/Canopy/issues/6
+    // Got a repeatable scenario by running this as a stress test with many iterations.
+    // A fix is to delay the status change notifications in the mock stream by a small amount.
+    // Not great because it makes the tests slower. Better solution would be to somehow ensure
+    // that the status deliveries are never out-of-order in ReplayingMockCKContainer.
     let container = ReplayingMockCKContainer(
       operationResults: [
         .accountStatus(.init(status: .available, error: nil)),
@@ -98,12 +104,13 @@ final class ContainerAPITests: XCTestCase {
     
     let containerAPI = CKContainerAPI(container: container, accountChangedSequence: .mock(elementsToProduce: 3))
     var statuses: [CKAccountStatus] = []
-    let stream = try? await containerAPI.accountStatusStream.get()
-    for await status in stream! {
+    let stream = try! await containerAPI.accountStatusStream.get()
+    for await status in stream {
       statuses.append(status)
       if statuses.count == 3 { break }
     }
-    XCTAssertEqual(statuses, [.available, .noAccount, .restricted])
+    let expectedStatuses: [CKAccountStatus] = [.available, .noAccount, .restricted]
+    XCTAssertEqual(statuses, expectedStatuses)
   }
   
   func test_accountStatus_twoStreams() async {

